Revit_Core_Adapter/AdapterActions/Push.cs

```diff
@@ -102,24 +103,68 @@ public override List<object> Push(IEnumerable<object> objects, string tag = "",
                 }
             }
 
-            // Push the objects
-            string transactionName = "BHoM Push " + pushType;
+            // First push everything that is not a Revit assembly
+            List<IBHoMObject> nonAssemblies = objectsToPush.Where(x => !(x is Assembly)).ToList();
+            List<IBHoMObject> pushed = PushToRevit(document, nonAssemblies, pushType, pushConfig, "BHoM Push " + pushType);
+
+            // Assemblies are being pushed separately in a few steps due to Revit API limitations in this field
+            List<IBHoMObject> revitAssemblies = objectsToPush.Where(x => x is Assembly).ToList();
+            if (revitAssemblies.Count != 0)
+            {
+                // Push assembly members first
+                List<IBHoMObject> pushedMembers = PushToRevit(document, revitAssemblies.SelectMany(x => ((Assembly)x).AssemblyMembers()), pushType, pushConfig, "BHoM Push " + pushType);
+
+                // Create the assemblies - no changes applied to the newly created assemblies here due to Revit API limitations
+                // Warnings are being suppressed to avoid all assembly-related warnings
+                UIControlledApplication.ControlledApplication.FailuresProcessing += ControlledApplication_FailuresProcessing;
+                List<IBHoMObject> createdAssemblies = new List<IBHoMObject>();
+                if (pushType == PushType.CreateOnly || pushType == PushType.CreateNonExisting || pushType == PushType.DeleteThenCreate || pushType == PushType.UpdateOrCreateOnly)
+                    createdAssemblies.AddRange(PushToRevit(document, revitAssemblies, pushType, pushConfig, "BHoM Push Create Assemblies"));
```

Comment:
I would personally use curly brackets here because both the condition and body of the `if` are long. Totally up to you of course!

Reply:
Following the offline discussion, let's stick to what we have for now, but a valid point regarding the length of code under `if` 
 - something to look info and possibly blanket cover across the entire code base 👍